packages/echo-base/src/module/utils.ts

```diff
@@ -1,19 +1,65 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-import { AppMetadata, SingleApp } from '../types/module';
+import { EchoModuleApiCreator, ModuleRequester } from '../types/creators';
+import { EchoModule, ModuleMetaData } from '../types/module';
 
+/**
+ *
+ *
+ * @export
+ * @param {string} id
+ */
 export function removeElementById(id: string): void {
     document.getElementById(id)?.remove();
 }
 
-export function isfunc(f: any): f is Function {
+/**
+ * Helper function for Checking if function is present.
+ *
+ * @param {EchoModuleApiCreator} createModuleApi
+ * @return {*}  {boolean}
+ */
+export function checkFunction(func: EchoModuleApiCreator | ModuleRequester, errorMessage: string): boolean {
+    if (!isfunc(func)) {
+        console.warn(errorMessage);
+        return false;
+    }
+
+    return true;
+}
+
+/**
+ *
+ *
+ * @export
+ * @param {*} f
+ * @return {*}  {f is Function}
+ */
+export function isfunc(f: unknown): f is Function {
     return typeof f === 'function';
 }
 
-export function createEmptyModule(meta: AppMetadata): SingleApp {
+/**
+ *
+ *
+ * @export
+ * @param {ModuleMetaData} meta
+ * @return {*}  {SingleApp}
+ */
+export function createEmptyApp(meta: ModuleMetaData): EchoModule {
     return {
         ...meta,
         setup(): void {
             // Empty module
         }
     };
 }
+
+export function isProduction(): boolean {
+    return process.env.NODE_ENV === 'production';
+}
+
+export function filterModulesByEnvironment(modules: EchoModule[], isProduction?: () => boolean): EchoModule[] {
```

Comment:
kan renames til mer forklarende, feks: filterExcludePrivateModulesInProduction. Jeg var litt forvirret at isProduction er optional, den bør ikke være med alltid? Og om det isåfall er nødvendig å ha den om inArgument siden den er definert rett ovenfor